qa-tests-backend/src/test/groovy/NetworkBaselineTest.groovy

```diff
@@ -277,57 +288,61 @@ class NetworkBaselineTest extends BaseSpecification {
         assert serverBaseline
 
+        // Add a client deployment
+        def beforeClientDeploymentCreate = System.currentTimeSeconds()
+        batchCreate([BASELINED_USER_CLIENT_DEP])
+        def justAfterClientDeploymentCreate = System.currentTimeSeconds()
+
+        def baselinedClientDeploymentID = BASELINED_USER_CLIENT_DEP.deploymentUid
+        assert baselinedClientDeploymentID != null
+        log.info "Client deployment: ${baselinedClientDeploymentID}"
+
+        def clientDeployment = DEPLOYMENTS.find { it.name == BASELINED_USER_CLIENT_DEP_NAME }
+
+        assert retryUntilTrue({
+            // Trigger http traffic to tests service deployment to ensure
+            // StackRox would see it and we can later include it in the baseline.
+            orchestrator.execInContainer(clientDeployment,
+                    "curl --insecure --head --request GET --max-time 5 http://${USER_DEP_NAME}")
+            return NetworkGraphUtil.checkForEdge(baselinedClientDeploymentID, serverDeploymentID)
+                    .any { it.targetID == serverDeploymentID}
+        }, 15)
+
+        // Grab the network baseline for the client.
         def baselinedClientBaseline = NetworkBaselineService.getNetworkBaseline(baselinedClientDeploymentID)
         assert baselinedClientBaseline
 
-        assert NetworkGraphUtil.checkForEdge(baselinedClientDeploymentID, serverDeploymentID, null, 180)
-
-        // Waiting on it to come out of observation.
-        serverBaseline = evaluateWithRetry(30, 4) {
-            def baseline = NetworkBaselineService.getNetworkBaseline(serverDeploymentID)
-            def now = System.currentTimeSeconds()
-            if (baseline.getPeersCount() == 0 && baseline.getObservationPeriodEnd().getSeconds() > now) {
-                throw new RuntimeException(
-                    "No peers in baseline for deployment ${serverDeploymentID} yet. Baseline is ${baseline}"
-                )
-            }
-            return baseline
-        }
+        // Grab a fresh copy of the serverBaseline after the client connection has been added.
+        serverBaseline = NetworkBaselineService.getNetworkBaseline(serverDeploymentID)
 
-        baselinedClientBaseline = NetworkBaselineService.getNetworkBaseline(baselinedClientDeploymentID)
+        log.info "Server Baseline: ${serverBaseline}"
+        log.info "Client Baseline: ${baselinedClientBaseline}"
 
         then:
         "Validate user requested server baseline"
         // The client->server connection should be baselined since the client as the
         // connection occurred during the observation window.
         validateBaseline(serverBaseline, beforeDeploymentCreate, justAfterDeploymentCreate,
-            [new Tuple2<String, Boolean>(baselinedClientDeploymentID, true)])
-        validateBaseline(baselinedClientBaseline, beforeDeploymentCreate, justAfterDeploymentCreate,
-            [new Tuple2<String, Boolean>(serverDeploymentID, false)]
```

Comment:
Why not validating client baseline any more?

Reply:
updated